pkg/operation/botanist/controlplane.go

```diff
@@ -1054,64 +1058,101 @@ func (b *Botanist) DeployETCD(ctx context.Context) error {
+		common.ETCDBackupRestoreImageName,
+	)
 	if err != nil {
 		return err
 	}
 
+	values["prefix"] = common.GenerateBackupEntryName(b.Shoot.Info.Status.TechnicalID, b.Shoot.Info.Status.UID)
+
 	for _, role := range []string{common.EtcdRoleMain, common.EtcdRoleEvents} {
-		etcd["role"] = role
+		values["role"] = role
+		values["backupEnabled"] = false
+		values["provider"] = ""
 		if role == common.EtcdRoleMain {
 			// etcd-main emits extensive (histogram) metrics
-			etcd["metrics"] = "extensive"
+			values["metrics"] = "extensive"
+			if b.Seed.Info.Spec.Backup != nil {
+				values["provider"] = b.Seed.Info.Spec.Backup.Provider
+				values["backupEnabled"] = true
+				secret := &corev1.Secret{}
+				if err := b.K8sSeedClient.Client().Get(ctx, kutil.Key(b.Shoot.SeedNamespace, common.BackupSecretName), secret); err != nil {
+					return err
+				}
+				values["container"] = string(secret.Data[common.BackupBucketName])
+			}
 		}
 
 		foundEtcd := true
-		statefulset := &appsv1.StatefulSet{}
-		if err := b.K8sSeedClient.Client().Get(ctx, kutil.Key(b.Shoot.SeedNamespace, fmt.Sprintf("etcd-%s", role)), statefulset); err != nil && !apierrors.IsNotFound(err) {
+		etcd := &druidv1alpha1.Etcd{}
+		if err := b.K8sSeedClient.Client().Get(ctx, kutil.Key(b.Shoot.SeedNamespace, fmt.Sprintf("etcd-%s", role)), etcd); client.IgnoreNotFound(err) != nil {
 			return err
 		} else if apierrors.IsNotFound(err) {
 			foundEtcd = false
 		}
 
-		if foundEtcd && hvpaEnabled {
+		foundStatefulset := true
+		statefulset := &appsv1.StatefulSet{}
+		if foundEtcd {
+			if etcd.Status.Etcd.Name != "" {
+				if err := b.K8sSeedClient.Client().Get(ctx, kutil.Key(b.Shoot.SeedNamespace, etcd.Status.Etcd.Name), statefulset); client.IgnoreNotFound(err) != nil {
+					return err
+				} else if apierrors.IsNotFound(err) {
+					foundStatefulset = false
+				}
+			}
+		}
+
+		if foundStatefulset && hvpaEnabled {
 			// etcd is already created AND is controlled by HVPA
 			// Keep the "resources" as it is.
 			for k := range statefulset.Spec.Template.Spec.Containers {
 				v := &statefulset.Spec.Template.Spec.Containers[k]
 				if v.Name == "etcd" {
-					etcd["etcdResources"] = v.Resources.DeepCopy()
+					values["etcdResources"] = v.Resources.DeepCopy()
+					break
+				} else if v.Name == "backup-restore" {
+					values["sidecarResources"] = v.Resources.DeepCopy()
 					break
 				}
 			}
 		}
 
 		if b.Shoot.HibernationEnabled {
-			// capture the old statefulset state
-			statefulset := &appsv1.StatefulSet{}
-			if err := client.IgnoreNotFound(b.K8sSeedClient.Client().Get(ctx, kutil.Key(b.Shoot.SeedNamespace, fmt.Sprintf("etcd-%s", role)), statefulset)); err != nil {
-				return err
-			}
-
 			// Restore the replica count from capture statefulset state.
-			etcd["replicas"] = 0
-			if statefulset.Spec.Replicas != nil {
-				etcd["replicas"] = *statefulset.Spec.Replicas
+			values["replicas"] = 0
+			if foundStatefulset && statefulset.Spec.Replicas != nil {
```

Comment:
```suggestion
			if foundEtcd && etcd.Spec.Replicas != nil {
			    values["replicas"] = *etcd.Spec.Replicas
			  
```

Reply:
Can you PTAL once again? The logic here. @ggaurav10

Reply:
Done